Desktop_Interface/isobuffer.cpp

```diff
@@ -2,391 +2,364 @@
-        if (gain_log == -1) buffer[i] *= 2;
-        else buffer[i] /= 2;
-    }
+	qDebug() << "Buffer shifted by" << gain_log;
+	for (int i=0; i<bufferEnd; i++)
+	{
+		if (gain_log == -1) buffer[i] *= 2;
+		else buffer[i] /= 2;
+	}
 }
 
-
+// NOTE: glitch insert is a no-op
 void isoBuffer::glitchInsert(short type)
 {
-
 }
 
-void isoBuffer::enableFileIO(QFile *file, int samplesToAverage, qulonglong max_file_size){
+void isoBuffer::enableFileIO(QFile* file, int samplesToAverage, qulonglong max_file_size)
+{
 
-    //Open the file
-    file->open(QIODevice::WriteOnly);
-    currentFile = file;
+	//Open the file
+	file->open(QIODevice::WriteOnly);
+	currentFile = file;
 
-    //Add the header
-    char headerLine[256];
-    sprintf(headerLine, "EspoTek Labrador DAQ V1.0 Output File\nAveraging = %d\nMode = %d\n", samplesToAverage, virtualParent->driver->deviceMode);
-    currentFile->write(headerLine);
+	//Add the header
+	char headerLine[256];
+	sprintf(headerLine, "EspoTek Labrador DAQ V1.0 Output File\nAveraging = %d\nMode = %d\n", samplesToAverage, virtualParent->driver->deviceMode);
+	currentFile->write(headerLine);
 
-    //Set up the isoBuffer for DAQ
-    fileIO_maxIncrementedSampleValue = samplesToAverage;
-    fileIO_max_file_size = max_file_size;
-    fileIO_sampleCount = 0;
-    fileIO_numBytesWritten = 0;
-    average_sample_temp = 0;
+	//Set up the isoBuffer for DAQ
+	fileIO_maxIncrementedSampleValue = samplesToAverage;
+	fileIO_max_file_size = max_file_size;
+	fileIO_sampleCount = 0;
+	fileIO_numBytesWritten = 0;
+	average_sample_temp = 0;
 
-    //Enable DAQ
-    fileIOEnabled = true;
+	//Enable DAQ
+	fileIOEnabled = true;
 
-    qDebug("File IO enabled, averaging %d samples, max file size %uMB", samplesToAverage, max_file_size/1000000);
-    qDebug() << max_file_size;
-    return;
+	qDebug("File IO enabled, averaging %d samples, max file size %lluMB", samplesToAverage, max_file_size/1000000);
+	qDebug() << max_file_size;
+	return;
 }
 
-void isoBuffer::disableFileIO(){
-    fileIOEnabled = false;
-    currentColumn = 0;
-    currentFile->close();
-    return;
+void isoBuffer::disableFileIO()
+{
+	fileIOEnabled = false;
+	currentColumn = 0;
+	currentFile->close();
+	return;
 }
 
-double isoBuffer::sampleConvert(short sample, int TOP, bool AC){
+double isoBuffer::sampleConvert(short sample, int TOP, bool AC) const
+{
 
-    double scope_gain = (double)(virtualParent->driver->scopeGain);
-    double voltageLevel;
+	double scope_gain = (double)(virtualParent->driver->scopeGain);
+	double voltageLevel;
 
-    voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
-    if (virtualParent->driver->deviceMode != 7) voltageLevel += voltage_ref;
-    #ifdef INVERT_MM
-        if(virtualParent->driver->deviceMode == 7) voltageLevel *= -1;
-    #endif
+	voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
+	if (virtualParent->driver->deviceMode != 7) voltageLevel += voltage_ref;
+#ifdef INVERT_MM
```

Comment:
Haha, I'm sorry for this #define.  Probably threw you a bit.  There was a particular old build (possibly even on a single board prototype) where the multimeter samples had their sign inverted by mistake.